text/0000-cache-types.md

```diff
@@ -0,0 +1,82 @@
+# Meta
+[meta]: #meta
+- Name: Cache Types
+- Start Date: 2020-06-26
+- Author(s): [Javier Romero](https://github.com/jromero)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Allow for buildpacks to specify what cache layers can be shared across applications.
+
+# Motivation
+[motivation]: #motivation
+
+In order to provide a better experience, platforms such as `pack` could provide a way to share cache across multiple applications. In particular, this is a frequently requested feature ([pack#340](https://github.com/buildpacks/pack/issues/340), [pack#652](https://github.com/buildpacks/pack/issues/652), [slack](https://buildpacks.slack.com/archives/C94UJCNV6/p1588718688412300), [slack](https://buildpacks.slack.com/archives/CD61YAG69/p1586370336098700)) for similar language family apps.
+
+Use Case:
+```
+Given I build a Java app A
+When I build a Java app B
+Then I expect similar maven dependencies not to have to be downloaded again
+```
+
+In order to solve this use case in a safe manner, the platform needs to know what cache it can share with other applications versus what cache is specific to a single application. This information is primarily known by the buildpacks.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Buildpacks will provide a mechanism to denote a cached layer as something that can be shared by associating a "type" to the cache.
```

Comment:
If I understand correctly, the buildpack author will decide what type of cache is going to be used, right?
I have a few questions about it:
1. Will the author be able to use both types of cache (each one for different purpose)?
2. If the buildpack author is using cache type x, does it make sense that the buildpack user won't want to use this type of cache? If yes, what will happen in this scenario? (I can't think about such real world scenario but I'm not familiar enough with our use cases)
3. What will happen if the buildpack author doesn't specify the cache type? I assume the default cache is `app-specific`, right?

Reply:
1. Yes, the buildpack author would be able to set a cache type per cache layer.
2. This is an interesting question. This would ultimately be up to the platforms to determine what level of control they want to provide to the end user. As you mention, it's hard to think of a case where this would be necessary since this is ultimately just creating a distinction and not so much changing any functionality.
3. `app-specific` would be the default (I'll update the doc).